File: cp_old/learn_practice/modulus_exponentiation.py
def modExp(a, n):
    if(n > 0):
        if((n & 1) == 1):
            return (a * modExp(a, n-1))
        else:
            return modExp(a**2, n >> 1)
    return 1


print(modExp(5, 5))

File: cp_old/learn_practice/modulus_exponentiation.cpp
#include <iostream>
#include <math.h>
using namespace std;

int modExp(int a, int n)
{
    if (n > 0)
    {
        if ((n & 1) == 1)
            return a * modExp(a, n - 1);
        return modExp(pow(a, 2), n >> 1);
    }
    return 1;
}
int main()
{
    cout << "Modulus Exponentiation result by recursion : " << modExp(5, 4) << endl;
    return 0;
}